tools/update_es_based_on_url_keyword.py

```diff
@@ -0,0 +1,172 @@
+
+    - URL_KEYWORD_TO_UPDATE (str): the keyword we seek to update the ES
+    - DELETE_ARTICLE_NEW_TOPICS (bool): if we should delete the current new topics for
+    the articles containing the URL_KEYWORD_TO_UPDATE. e.g. we note that there is a 
+    keyword wrongly associated with a specific topic.
+    - RECALCULATE_TOPICS (bool): if the topics for the articles should be recalculated.
+    Let's say two keywords have the same topic, then the article would loose it's topic 
+    despite still being categorized by another keyword. E.g. 'vejret' and 'klima' are 
+    associated with 'Technology & Science'. If we delete topics associated, with 'vejret' 
+    all articles with 'vejret' will loose that topic. This would be incorrect for all 
+    those that continue to have 'klima' as a keyword. For this reason, 
+    RECALCULATE_TOPICS should be true. This setting MUST BE true, in case of an update,
+    e.g. we adde a new mapping to one of the keywords.
+    - RE_INDEX_ONLY_ARTICLES_IN_ES (bool): if the articles re-index are only those 
+    that were already in ES.
+    - ITERATION_STEP (int): number of articles to index in each loop. 
+"""
+
+URL_KEYWORD_TO_UPDATE = "vejret"
+DELETE_ARTICLE_NEW_TOPICS = True
+RECALCULATE_TOPICS = True
+RE_INDEX_ONLY_ARTICLES_IN_ES = True
+ITERATION_STEP = 1000
+
+
+# coding=utf-8
+from zeeguu.core.elastic.indexing import (
+    create_or_update_bulk_docs,
+)
+
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import bulk, scan
+import zeeguu.core
+from zeeguu.core.model import Article
+from datetime import datetime
+from sqlalchemy.orm.exc import NoResultFound
+from zeeguu.api.app import create_app
+
+from zeeguu.core.model import (
+    ArticleUrlKeywordMap,
+    UrlKeyword,
+    NewArticleTopicMap,
+    NewTopic,
+)
+from zeeguu.core.model.article import article_topic_map
+from zeeguu.core.elastic.settings import ES_ZINDEX, ES_CONN_STRING
+import numpy as np
+from tqdm import tqdm
+
+app = create_app()
+app.app_context().push()
+
+
+es = Elasticsearch(ES_CONN_STRING)
+db_session = zeeguu.core.model.db.session
+print(es.info())
+
+
+def main():
+    def fetch_articles_by_id(id_list, recalculate_topic=False):
+        for i in id_list:
+            try:
+                article = Article.find_by_id(i)
+                if not article:
+                    print(f"Skipped for: '{i}', article not in DB.")
+                    continue
+                if recalculate_topic:
+                    article.recalculate_topics_from_url_keywords(db_session)
+                    db_session.commit()
+                yield (article)
+            except NoResultFound:
+                print(f"fail for: '{i}'")
+            except Exception as e:
+                print(f"fail for: '{i}', {e}")
+
+    def gen_docs(articles_w_topics):
+        for article in articles_w_topics:
+            try:
+                yield create_or_update_bulk_docs(article, db_session)
```

Comment:
why is this called `bulk_docs`? it seems to work on a single article.

Reply:
It is an iterator that works for each article and creates a json to work for the bulk function, so the bulk here means that it's to be used by the bulk functions, not necessary that is taking a bulk of articles. I renamed it to: 'create_or_update_doc_for_bulk' maybe that's a bit more clear?